lib/router/transition.js

```diff
@@ -58,7 +70,7 @@ function Transition(router, intent, state, error) {
       this.pivotHandler = handlerInfo.handler;
     }
 
-    this.sequence = Transition.currentSequence++;
+    this.sequence = router.currentSequence++;
     this.promise = state.resolve(checkForAbort, this)['catch'](function(result) {
```

Comment:
So to be clear, right now the best thing for me to do is to refactor that function in the transition constructor to a prototype method, so that it doesn't risk closing over the arguments passed in to the transition constructor?

FWIW I hadn't thought that it would close over variables that you don't explicitly use in the function body, however that's based on chrome's behaviour and not a deep understanding of the various js VMs ember supports